### ISSUE TITLE
OmniGenCache makes Omnitel CUDA is there an option to no use it.

### ISSUE THREAD

**Issue by Vargol** [VectorSpaceLab/OmniGen#91]
looking at the __call__ code I can't see a way to avoid the OminGenCache which is full of CUDA only code
As you've got code in the pipeline to detect only runtimes like mps is there a way to avoid the cache

**Comment by staoxiao** [comment]
Hi, @Vargol , you can delete the two lines: https://github.com/VectorSpaceLab/OmniGen/blob/main/OmniGen/scheduler.py#L13-L14 to run the code in mps device (and set offload_kv_cache=False). However, running it on MAC is very slow in our experiments, and we don't recommend this behavior.

**Comment by bghira** [comment]
don't need those to be deleted. just:

```py
                    pipeline_kwargs["use_kv_cache"] = (
                        False if torch.backends.mps.is_available() else True
                    )
                    pipeline_kwargs["offload_kv_cache"] = (
                        False if torch.backends.mps.is_available() else True
                    )
```

both of those have to be disabled.

**Comment by Vargol** [comment]
> don't need those to be deleted. just:
> 
> ```python
>                     pipeline_kwargs["use_kv_cache"] = (
>                         False if torch.backends.mps.is_available() else True
>                     )
>                     pipeline_kwargs["offload_kv_cache"] = (
>                         False if torch.backends.mps.is_available() else True
>                     )
> ```
> 
> both of those have to be disabled.

Thanks for that, ran at 32 s/i without swapping on my 10 GPU core  24Gb M3 for a straight forward text 2 image run, which about the same speed as Flux thought InvokeAI (although that's also using a GGUG Q8 model to avoid swap).

Are the 50 steps necessary or just a default value ?

**Comment by bghira** [comment]
lt'll "work" at 8 steps 👍

**Comment by karlbit9000** [comment]
> don't need those to be deleted. just:
> 
> ```python
>                     pipeline_kwargs["use_kv_cache"] = (
>                         False if torch.backends.mps.is_available() else True
>                     )
>                     pipeline_kwargs["offload_kv_cache"] = (
>                         False if torch.backends.mps.is_available() else True
>                     )
> ```
> 
> both of those have to be disabled.

Please, where I put these ?

**Comment by NatanFreeman** [comment]
@karlbit9000 I believe at `OmniGen/scheduler.py` lines 13-14

**Comment by Vargol** [comment]
> > don't need those to be deleted. just:
> > ```python
> >                     pipeline_kwargs["use_kv_cache"] = (
> >                         False if torch.backends.mps.is_available() else True
> >                     )
> >                     pipeline_kwargs["offload_kv_cache"] = (
> >                         False if torch.backends.mps.is_available() else True
> >                     )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > both of those have to be disabled.
> 
> Please, where I put these ? (I'm running Pinokio)

I don't know about Pinokio  but you but them it the script to run Omnigen

```py
import torch 
from OmniGen import OmniGenPipeline

pipeline_kwargs = {};

pipeline_kwargs["use_kv_cache"] = (
   False if torch.backends.mps.is_available() else True
)

pipeline_kwargs["offload_kv_cache"] = (
    False if torch.backends.mps.is_available() else True
)

pipe = OmniGenPipeline.from_pretrained("Shitao/OmniGen-v1")

# Text to Image
images = pipe(
    prompt="A curly-haired man in a red shirt is drinking tea.", 
    height=1024, 
    width=1024, 
    guidance_scale=2.5,
    seed=0,
    num_inference_steps=20,
    **pipeline_kwargs
)
images[0].save("example_t2i.png")  # save output PIL Image
```

If you know you're only going to run on macOS skip the fancy kwarg stuff and hard code it e.g.
```py
import torch
from OmniGen import OmniGenPipeline

pipe = OmniGenPipeline.from_pretrained("Shitao/OmniGen-v1")

# Text and Image Prompting
images = pipe(
    prompt="A man in a black shirt is reading a book. The man is the right man in <img><|image_1|></img>.",
    input_images=["./imgs/test_cases/two_man.jpg"],
    height=1024,
    width=1024,
    guidance_scale=2.5,
    img_guidance_scale=1.6,
    seed=0,
    use_kv_cache=False,
    offload_kv_cache=False
)
images[0].save("example_ti2i.png")  # save output PIL image
```
